File: fsharp/anagram/Anagram.fs
module Anagram

let findAnagrams (sources: string list) (target: string): string list =
    let sort (str: string) = str.ToLower() |> Seq.sort |> Seq.toArray

    sources
    |> List.filter (fun x -> x.ToLower() <> target.ToLower()
                             && sort x = sort target)

File: clojure/anagram/src/anagram.clj
(ns anagram
  (:require [clojure.string :as str]))

(defn- normalise [word]
  (sort (str/lower-case word)))

(defn anagrams-for [word prospect-list]
  (->> prospect-list
       (filter #(and (not= (str/lower-case %) (str/lower-case word))
                     (= (normalise %) (normalise word))))))
